### PR TITLE
Scan all bucket dbs in storage::distributor::SimpleMaintenanceScanner.

### PR DESCRIPTION
**Pull request by toregge**
@geirst, @vekterli: please review

### CODE REVIEW

File: storage/src/tests/distributor/simplemaintenancescannertest.cpp [hunk 1]
```diff
@@ -80,7 +94,27 @@ SimpleMaintenanceScannerTest::testPrioritizeSingleBucket()
     addBucketToDb(1);
     std::string expected("PrioritizedBucket(Bucket(BucketSpace(0x0000000000000000), BucketId(0x4000000000000001)), pri VERY_HIGH)\n");
 
-    CPPUNIT_ASSERT(!_scanner->scanNext().isDone());
+    auto scanResult = _scanner->scanNext();
+    CPPUNIT_ASSERT(!scanResult.isDone());
+    CPPUNIT_ASSERT_EQUAL(scanResult.getBucketSpace().getId(), makeBucketSpace().getId());
```
Comment (vekterli) [inline storage/src/tests/distributor/simplemaintenancescannertest.cpp]: 1st argument to macro is the expected value, so flipping the arguments around will ensure that correct expected/actual values are printed.

File: storage/src/tests/distributor/simplemaintenancescannertest.cpp [hunk 2]
```diff
@@ -80,7 +94,27 @@ SimpleMaintenanceScannerTest::testPrioritizeSingleBucket()
...
+{
+    document::BucketSpace bucketSpace(4);
+    _bucketSpaceRepo->add(bucketSpace, std::make_unique<DistributorBucketSpace>());
+    _scanner->reset();
+    addBucketToDb(bucketSpace, 1);
+    std::string expected("PrioritizedBucket(Bucket(BucketSpace(0x0000000000000004), BucketId(0x4000000000000001)), pri VERY_HIGH)\n");
+
+    auto scanResult = _scanner->scanNext();
+    CPPUNIT_ASSERT(!scanResult.isDone());
+    CPPUNIT_ASSERT_EQUAL(scanResult.getBucketSpace().getId(), bucketSpace.getId());
```
Comment (vekterli) [inline storage/src/tests/distributor/simplemaintenancescannertest.cpp]: Same as above

**Review by vekterli — APPROVED** [review]
:+1: only a very minor comment on assertion macro argument order

**Review by vekterli — APPROVED** [review]
:+1:

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**